src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -357,6 +357,66 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
             return expandedBytes;
         }
 
+        private static unsafe string UrlEncodeInternal(string originalString)
```

Comment:
This should not be separate function since it is called just from one place and the caller has very little code it in. Fold it into the caller.